src/dosbox.cpp

```diff
@@ -183,34 +183,18 @@ void increaseticks() { //Make it return ticksRemain and set it in the function a
 		return;
 	}
 
-	static Bit32s lastsleepDone = -1;
-	static Bitu sleep1count = 0;
-
 	const auto ticksNew = GetTicks();
 	ticksScheduled += ticksAdded;
 	if (ticksNew <= ticksLast) { //lower should not be possible, only equal.
 		ticksAdded = 0;
 
-		if (!CPU_CycleAutoAdjust || CPU_SkipCycleAutoAdjust || sleep1count < 3) {
-			delay_fn(1);
-		} else {
-			/* Certain configurations always give an exact sleepingtime of 1, this causes problems due to the fact that
-			   dosbox keeps track of full blocks.
-			   This code introduces some randomness to the time slept, which improves stability on those configurations
-			 */
-			static const Bit32u sleeppattern[] = { 2, 2, 3, 2, 2, 4, 2 };
-			static Bit32u sleepindex = 0;
-			if (ticksDone != lastsleepDone) sleepindex = 0;
-			delay_fn(sleeppattern[sleepindex++]);
-			sleepindex %= sizeof(sleeppattern) / sizeof(sleeppattern[0]);
-		}
-		auto timeslept = GetTicksSince(ticksNew);
-		// Count how many times in the current block (of 250 ms) the time slept was 1 ms
-		if (CPU_CycleAutoAdjust && !CPU_SkipCycleAutoAdjust && timeslept == 1) sleep1count++;
-		lastsleepDone = ticksDone;
+		const auto start = GetTicksUs();
+		while(GetTicksUsSince(start) <= 1000);	
+		
+		//const auto timeslept = GetTicksSince(ticksNew);
```

Comment:
This flies under the radar without a loop body (but it's very important); maybe make it a bit more prominent?